app/src/editor/canvas/components/DraggableCanvasWindow.jsx

```diff
@@ -35,74 +36,127 @@ export const Toolbar = ({ children, className }) => (
     </div>
 )
 
-class DraggableCanvasWindow extends React.Component {
-    static Dialog = Dialog
-    static Title = Title
-    static Toolbar = Toolbar
-
-    static defaultProps = {
-        start: {
-            x: 0,
-            y: 0,
-        },
-    }
-
-    state = {
-        // Whether or not we are currently dragging.
+export const DraggableCanvasWindow = (props) => {
+    const [layout, setLayoutState] = useState({
         dragging: false,
+        resizing: false,
         // Current transform x and y.
-        clientX: this.props.start.x,
-        clientY: this.props.start.y,
-    }
-
-    onDragStart = () => {
-        this.setState({
+        clientX: props.position.x,
+        clientY: props.position.y,
+        width: props.size.width,
+        height: props.size.height,
```

Comment:
Whenever we copy props to state, it's a big invitation for subtle bugs. I wonder if there's some hidden bugs here if props.position/size gets updated to something different than what's held in state e.g. via undo/redo

The pattern I've been using is: use the value from props until we absolutely need to take control of the component e.g. use value from state only during drag, use the value from props otherwise

Reply:
FYI this exact thing was causing a bunch of issues with the code editor's `code` value, see https://github.com/streamr-dev/streamr-platform/pull/427

And we've seen similar issues with same underlying cause (copying props to state but never updating state) for at least the colour picker & Map ports.

Reply:
I changed it so that the state value is only used when dragging / resizing.

Reply:
👍